src/mono/mono/mini/mini-llvm.c

```diff
@@ -6119,7 +6119,7 @@ process_bb (EmitContext *ctx, MonoBasicBlock *bb)
 				gboolean is_simd = mini_class_is_simd (ctx->cfg, mono_class_from_mono_type_internal (sig->ret));
 				switch (linfo->ret.storage) {
 				case LLVMArgNormal: src_in_reg = TRUE; break;
-				case LLVMArgVtypeInReg: case LLVMArgVtypeAsScalar: src_in_reg = is_simd; break;
+				case LLVMArgVtypeInReg: case LLVMArgVtypeAsScalar: case LLVMArgWasmVtypeAsScalar: src_in_reg = is_simd; break;
```

Comment:
```suggestion
				case LLVMArgVtypeInReg:
				case LLVMArgVtypeAsScalar:
				case LLVMArgWasmVtypeAsScalar:
					src_in_reg = is_simd;
					break;
```

follow-up idea: this function with a ton of nested switch statements is hard to think about.  Let's make helper methods.

Reply:
https://github.com/dotnet/runtime/issues/101412